### PR TITLE
Web UI code separation and improvements

### PR DESCRIPTION
**Pull request by chappjc**
This PR adds:

- web.go, containing a web ui type and it's associated functions.
- reloadsig.go and reloadsig_windows to allow reparsing templates on SIGUSR1, which allows templates to be reloaded without restarting the whole app.
- A LICENSE file (ISC).
- Check for no agendas when updating template info.

The cli flags commit is part of the other PR.

### CODE REVIEW

**Comment by chappjc** [comment]
@karamble I need this and PR https://github.com/decred/hardforkdemo/pull/61 to go in before I add both the stubbed client (fake JSON data) and the data storage system.  However, I don't have an easy way to test since the agendas are gone from testnet2.  Can this be tried on the testnet1 machine?

**Comment by chappjc** [comment]
@alexlyp Do you guys have a way to run a simnet voting scenario? hardforkdemo could latch on to that, if it exists.

**Comment by chappjc** [comment]
Alternatively I can instead PR into a devel branch until we're able to work out a simple testing method.  I'd like to get moving on the next steps, particularly the JSON client stub, but I don't want to pile PR on top of PR.

Or we can move forward with the understanding that this might not work as intended yet, but that will change when we've got a fake data input.

**Comment by karamble** [comment]
i expect a relaunch of the complete frontend as it currently got redesigned from scratch by the branding team, read more about it in the #hardforkwebsite slack room ...

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,17 @@
+ISC License
+
+Copyright (c) 2015-2017, The Decred developers
+Copyright (c) 2017, Jonathan Chappelow
+Copyright (c) 2014-2016, The btcsuite developers
```
Comment (alexlyp) [inline LICENSE]: You can remove btcsuite devs from this repo.

**Review by alexlyp — CHANGES_REQUESTED** [review]
Looks good, just need that license fix and that #61 fix

**Review by alexlyp — APPROVED** [review]

**Pull request merged**